src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxFactoryTests.cs

```diff
@@ -503,5 +503,20 @@ public void TestSpacingOnNullableDatetimeType()
             // no space between DateTime and ?
             Assert.Equal("class C\r\n{\r\n    DateTime? P\r\n    {\r\n    }\r\n}", syntaxNode.ToFullString());
         }
+
+        [Fact]
+        [WorkItem(21231, "https://github.com/dotnet/roslyn/issues/21231")]
+        public void TestSpacingOnTernary()
+        {
+            var syntaxNode = SyntaxFactory.ParseExpression("x is int? y: z").NormalizeWhitespace();
+
+            // space between int and ?
+            Assert.Equal("x is int ? y : z", syntaxNode.ToFullString());
+
+            var syntaxNode2 = SyntaxFactory.ParseExpression("x is DateTime? y: z").NormalizeWhitespace();
+
+            // space between DateTime and ?
+            Assert.Equal("x is DateTime ? y : z", syntaxNode2.ToFullString());
```

Comment:
Perhaps `x is int??y:z`, `x as object??y` as well.

Reply:
Added test. The `: z` portion gets parse as trivia, so I removed it.
Also added a helper method to dump syntax trees, like I recently added to VB test helpers.